### DISCUSSION TITLE
How to permanently delete the "Deleted repositories"

### DISCUSSION THREAD

**Discussion by jacksonyin** [community/community#134143] [New to GitHub] [answered]
I try github and build some test repositories, and deleted them.
but I can found them in settings-repositoies-'Deleted repositories'
I expect delete them permanently and never restore, how can I do?

**Comment by GabrieleTronchin** [comment, marked as answer]
Hi Jackson,

Deleted repository stay availbale to restore for 90 days, then it will be permanently deleted.

Here the docs:

[deleting-a-repository](https://docs.github.com/en/repositories/creating-and-managing-repositories/deleting-a-repository)

[restoring](https://docs.github.com/en/repositories/creating-and-managing-repositories/restoring-a-deleted-repository)

**Reply by HamnaShafiq** [reply]
That' s actually right and jakson it would be great if you mark the comment as answered if you get it.

**Comment by Sivamani-18** [comment]
1. **Delete Repositories via API:**
   - First, generate a personal access token with the appropriate permissions from GitHub.
   - Use a tool like `curl` to send a DELETE request for each repository.

```sh
curl -X DELETE -H "Authorization: token YOUR_GITHUB_TOKEN" https://api.github.com/repos/USERNAME/REPOSITORY_NAME
```
Replace `YOUR_GITHUB_TOKEN` with your actual GitHub token, `USERNAME` with your GitHub username, and `REPOSITORY_NAME` with the name of the repository.
